src/backend/mpi/swdsm.cpp

```diff
@@ -769,17 +769,20 @@ void * loadcacheline(void * x){
 
 		unsigned long offset = getOffset(lineAddr);
 		if(isPowerOf2((tempsharer)&invid) && tempsharer != id && prevsharer == 0){ //Other private. but may not have loaded page yet.
-			long ownid = (long)(tempsharer&invid);
-			unsigned long owner;
-			for(n=0; n<numtasks; n++){
-				if((unsigned)(1<<n)==(unsigned)ownid){
+			unsigned long ownid = tempsharer&invid; // remove own bit
+			static const unsigned long invalid_owner = static_cast<unsigned long>(-1);
```

Comment:
Readability shouldn't be an issue as the actual value is not important, it's just some value that signifies an invalid node ID. Just like any global constant or #define value.

I agree with making it static, but using namespace {} instead of static.